### ISSUE TITLE
About Missing-modality Generalization

### ISSUE THREAD

**Issue by misakamage** [donghao51/SimMMDG#10]
Hello! Thank you for releasing this great work!
I’m very interested in the Missing-modality Generalization mentioned in your paper. However, I noticed that the code for this part doesn't seem to be included in the released repository. If it has not been open-sourced yet, would you mind sharing some details on how you implemented this part?

**Comment by donghao51** [comment]
hello, thanks for your interests!

After training, we only fine-tune the translation module (**mlp_v2a** and **mlp_a2v**) and keep other parameters fixed. For fintuning, we use the loss:

```
a_emd_t = mlp_v2a(v_emd)
v_emd_t = mlp_a2v(audio_emd)
v2a_loss = torch.mean(torch.norm(a_emd_t-audio_emd, dim=1))
a2v_loss = torch.mean(torch.norm(v_emd_t-v_emd, dim=1))
loss = (v2a_loss + a2v_loss)/2
```

During inference, the code becomes:

```
a_emd_t = mlp_v2a(v_emd)
v_emd_t = mlp_a2v(audio_emd)
predict = mlp_cls(v_emd, audio_emd) # original prediction
predict_no_audio = mlp_cls(v_emd,a_emd_t) # audio translation
predict_no_v = mlp_cls(v_emd_t,audio_emd) # video translation
predict_no_audio_zero = mlp_cls(v_emd,torch.zeros_like(audio_emd).cuda()) # audio zero-filling
predict_no_v_zero = mlp_cls(torch.zeros_like(v_emd).cuda(),audio_emd) # video zero-filling
```
